app/src/test/java/com/example/concordia_campus_guide/ModelTests/BuildingTest.java

```diff
@@ -0,0 +1,265 @@
+    private float width = 68;
+    private float height = 68;
+    private float bearing = 34;
+    private String campus = "SGW";
+    private String buildingCode = "H";
+    private String Building_Long_Name = "Henry F. Hall Building";
+    private String address = "1455 De Maisonneuve West";
+    private List<String> departments = new ArrayList<String>(Arrays.asList(
+            "Geography, Planning and Environment",
+            "Political Science, Socialogy and Anthropology, Economics",
+            "School of Irish Studies"
+      ));
+    private List<String> services = new ArrayList<String>(Arrays.asList("Welcome Crew Office",
+            "DB Clarke Theatre",
+            "Dean of Students",
+            "Aboriginal Student Resource Centre",
+            "Concordia Student Union",
+            "IT Service Desk",
+            "Security Office",
+            "Student Success Centre",
+            "Mail Services",
+            "Archives",
+            "Career and Planning Services",
+            "Sexual Assault Resource Centre (SARC)"
+    ));
+
+    Coordinates cornerCoordinate1 = new Coordinates(45.497178,-73.57955);
+    Coordinates cornerCoordinate2 = new Coordinates(45.497708,-73.579035);
+    Coordinates cornerCoordinate3 = new Coordinates(45.497385,-73.578332);
+    Coordinates cornerCoordinate4 = new Coordinates(45.496832,-73.578842);
+    Coordinates cornerCoordinate5 = new Coordinates(45.497178,-73.57955);
+
+    List<Coordinates> listOfCornerCoordinates = new ArrayList<Coordinates>() {{
+        add(cornerCoordinate1);
+        add(cornerCoordinate2);
+        add(cornerCoordinate3);
+        add(cornerCoordinate4);
+        add(cornerCoordinate5);
+    }};
+
+    private ListOfCoordinates cornerCoordinates = new ListOfCoordinates(listOfCornerCoordinates);
+
+    @Before
+    public void init() {
+        defaultBuilding = new Building();
+        buildingWithArgs = new Building(centerCoordinates, availableFloors, width, height,
+                bearing, campus, buildingCode, Building_Long_Name,  address, departments,
+                services,  cornerCoordinates
+        );
+    }
+
+    @Test
+    public void getAvailableFloors() {
+        assertNull(defaultBuilding.getAvailableFloors());
+        assertEquals(availableFloors,buildingWithArgs.getAvailableFloors());
+    }
+
+    @Test
+    public void setAvailableFloors() {
+        ArrayList<String> updateAvailableFloors =  new ArrayList<String>(Arrays.asList("0", "1", "8", "9"));
+        assertEquals(availableFloors, buildingWithArgs.getAvailableFloors());
+        buildingWithArgs.setAvailableFloors(updateAvailableFloors);
+        assertNotEquals(availableFloors, buildingWithArgs.getAvailableFloors());
+        assertEquals(updateAvailableFloors, buildingWithArgs.getAvailableFloors());
+    }
+
+    @Test
+    public void getWidth() {
+        assertEquals(68, buildingWithArgs.getWidth(), 0.001);
```

Comment:
What is the 0.001 for?

Reply:
Because width is a double, it requires a delta to compare their equality

Reply:
ok thanks